actionpack/lib/action_dispatch/middleware/stack.rb

```diff
@@ -66,6 +66,10 @@ def call(env)
           @middleware.call(env)
         end
       end
+
+      def inspect # :nodoc:
+        "#<#{self.class.name}:#{'%#016x' % (object_id << 1)} @middleware=#{@middleware.inspect}>"
+      end
```

Comment:
What are the drawbacks of just passing through to `@middleware.inspect`?  i.e.:

```suggestion
      def inspect # :nodoc:
        @middleware.inspect
      end
```

Are there use cases where we want to inspect the middleware stack and see all the `ActionDispatch::MiddlewareStack::InstrumentationProxy` instances?

_Sidenote:_ Since Ruby 2.7 (I think), `'%#016x' % (object_id << 1)` does not yield the object's address.  The result is something like "0x00000000002df0" instead of "0x00007ff41537a788".  I don't think the difference is particularly important, but 16 hex digits might be overkill.

Reply:
> What are the drawbacks of just passing through to @middleware.inspect?

This would hide the `InstrumentationProxy` in the middleware stack. This could make debugging issues with middleware confusing (for example, stack traces won't match up with the inspected middleware) so my preference would be not to hide it.

> Sidenote: Since Ruby 2.7 (I think), '%#016x' % (object_id << 1) does not yield the object's address. The result is something like "0x00000000002df0" instead of "0x00007ff41537a788". I don't think the difference is particularly important, but 16 hex digits might be overkill.

I wasn't sure what to do here so followed the pattern of other inspect methods I found in Rails.